### PR TITLE
gerrit-queue: init at 0.0.1

### PR DESCRIPTION
**Pull request by flokli**
This adds gerrit-queue, a submit bot for gerrit, to nixpkgs.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/ge/gerrit-queue/package.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+  src = fetchFromGitHub {
+    owner = "flokli";
+    repo = "gerrit-queue";
+    rev = "0186dbde15c9b11dc17b422feb74c842f6fa605a";
+    hash = "sha256-zXB5vre/Vr7UOyeMnf2RCtMKm+v5RENH7kGPr/2o7mI=";
+  };
+  vendorHash = "sha256-+Ig4D46NphzpWKXO23Haea9EqVtpda8v9zLPJkbe3bQ=";
+
+  meta = with lib; {
```
Comment (MikaelFangel) [inline pkgs/by-name/ge/gerrit-queue/package.nix:18]: You're missing the meta.maintainers tag. How should maintain this package? (:

File: pkgs/by-name/ge/gerrit-queue/package.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    description = "Gerrit submit bot";
+    homepage = "https://github.com/tweag/gerrit-queue";
+    license = licenses.asl20;
```
Comment (MikaelFangel) [inline pkgs/by-name/ge/gerrit-queue/package.nix:21]:
```suggestion
    license = licenses.asl20;
    mainProgram = "gerrit-queue";
```

**Review by MikaelFangel — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ge/gerrit-queue/package.nix (hunk 1 above)
Comment (flokli) [reply pkgs/by-name/ge/gerrit-queue/package.nix:18]: fixed.

**Review by MikaelFangel — APPROVED** [review]
LGTM

File: pkgs/by-name/ge/gerrit-queue/package.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{ buildGoModule
+, lib
+, fetchFromGitHub
+}:
+
+buildGoModule {
+  pname = "gerrit-queue";
+  version = "unstable-2023-10-20";
```
Comment (zimbatm) [inline pkgs/by-name/ge/gerrit-queue/package.nix:8]: If it's good enough to be packaged, how about cutting a release as well?

File: pkgs/by-name/ge/gerrit-queue/package.nix [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    owner = "flokli";
+    repo = "gerrit-queue";
+    rev = "0186dbde15c9b11dc17b422feb74c842f6fa605a";
+    hash = "sha256-zXB5vre/Vr7UOyeMnf2RCtMKm+v5RENH7kGPr/2o7mI=";
+  };
+  vendorHash = "sha256-+Ig4D46NphzpWKXO23Haea9EqVtpda8v9zLPJkbe3bQ=";
+
+  meta = with lib; {
+    description = "Gerrit submit bot";
+    homepage = "https://github.com/tweag/gerrit-queue";
```
Comment (zimbatm) [inline pkgs/by-name/ge/gerrit-queue/package.nix:20]: Which one is the canonical version; tweag's or yours?

Comment (flokli) [reply pkgs/by-name/ge/gerrit-queue/package.nix]: Sorry, this is an old expression, the `tweag` one redirects to `flokli` actually. Updated it to `flokli`.

File: pkgs/by-name/ge/gerrit-queue/package.nix (hunk 3 above)
Comment (flokli) [reply pkgs/by-name/ge/gerrit-queue/package.nix:8]: There's some more things I'd like to address before cutting a release, like cleaning up the whole log/web interface story. I'd put it not in the critical path for the package recipe.

Comment (flokli) [reply pkgs/by-name/ge/gerrit-queue/package.nix]: I applied some small fixes in https://github.com/flokli/gerrit-queue/pull/16, and tagged a v0.0.1. PTAL.

**Comment by RaitoBezarius** [comment]
Result of `nixpkgs-review pr 265922` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gerrit-queue</li>
  </ul>
</details>

**Pull request merged**